File: app/src/main/java/kr/djspi/pipe01/dto/PipeShape.java
package kr.djspi.pipe01.dto;

import java.io.Serializable;

import lombok.Data;
import lombok.Getter;

@Data
public class PipeShape implements DataItem, Serializable {

    /**
     * '관로형태(shape)' 와 방향, 스펙 정보를 함께 다루는 클래스
     * T 분기형 관로의 경우 본관, 지관 구경이 달라질 수 있어 spec_sub 를 두고 사용한다.
     */
    private int id = -1;
    private int pipe_id = -1;
    private String shape;
    private String spec;
    private String spec_sub;

    @Getter
    @SuppressWarnings("NonAsciiCharacters")
    public enum PipeShapeEnum {

        직진형("직진형", "str"),
        T분기형("T분기형", "tbr"),
        엘보형("엘보형", "elb"),
        관말형("관말형", "end");

        private String name;
        private String code;

        PipeShapeEnum(String name, String code) {
            this.name = name;
            this.code = code;
        }

        public static String parsePipeShape(String pipeShape) {
            for (PipeShapeEnum shapeEnum : PipeShapeEnum.values()) {
                if (shapeEnum.name.equals(pipeShape)) {
                    return shapeEnum.code;
                }
            }
            return null;
        }
    }
}

File: app/src/main/java/kr/djspi/pipe01/dto/PipeShape.kt
package kr.djspi.pipe01.dto

import java.io.Serializable

@Suppress("NonAsciiCharacters")
data class PipeShape(var id: Int = -1) : DataItem, Serializable {

    var pipe_id: Int = -1
    var shape: String? = null
    var spec: String? = null
    var spec_sub: String? = null

    /**
     * 새 관로형태를 목록에 추가하려면 아래의 Enum 에 항목을 추가해준다 (추가 수정 불필요)
     */
    @Suppress("NonAsciiCharacters", "EnumEntryName", "unused")
    enum class PipeShapeEnum(internal var type: String, internal var code: String) {

        직진형("직진형", "str"),
        T분기형("T분기형", "tbr"),
        엘보형("엘보형", "elb"),
        엘보형135("엘보형(135º)", "elb135"),
        관말형("관말형", "end"),
        선택형("선택형", "none");

        companion object {
            fun parsePipeShape(pipeShape: String?): String? {
                values().forEach {
                    if (it.type == pipeShape) {
                        return it.code
                    }
                }
                return null
            }

            fun getSelectableTypes(): List<String> {
                val array = mutableListOf<String>()
                for (i in 0..values().size - 2) {
                    array.add(i, values()[i].type)
                }
                return array
            }
        }
    }
}
